executor/builder.go

```diff
@@ -5196,9 +5196,25 @@ func (b *executorBuilder) buildCompactTable(v *plannercore.CompactTable) Executo
 		return nil
 	}
 
+	// check partition
+	if v.PartitionNames != nil {
+		if v.TableInfo.Partition == nil {
+			b.err = errors.Errorf("table:%s is not a partition table, but user specify partition name list:%+v", v.TableInfo.Name.O, v.PartitionNames)
+			return nil
+		}
+		for _, partitionName := range v.PartitionNames {
+			partition := v.TableInfo.FindPartitionDefinitionByName(partitionName.L)
```

Comment:
We have already iterated and checked partition. How about just pass this partition to `CompactTableTiFlashExec` to avoid look up partition again?